include/circt/Dialect/RTL/RTLCombinatorial.td

```diff
@@ -323,6 +323,29 @@ def ArraySliceOp : RTLOp<"array_slice", [NoSideEffect]> {
   ];
 }
 
+def ArrayGetOp
+ : RTLOp<"array_get", [NoSideEffect]> {
+  let summary = "Get the value in an array at the specified index";
+  let arguments = (ins ArrayType:$input, RTLIntegerType:$index);
+  let results = (outs AnyType:$result);
```

Comment:
Instead of declaring the result as `AnyType`, it would be better to use a `TypesMatchWith` thing like `InOutElementConstraint` to constrain it here.  This will cause the verifier hook to get generated for you, and eliminate the need for the custom parser/printer.

You can do the same thing for the index width constraint.

Reply:
Nice!